setup.py

```diff
@@ -94,6 +94,7 @@
             "ruff~=0.7.0",
             "snowflake-connector-python[pandas,secure-local-storage]>=3.0.2",
             "sqlalchemy-stubs",
+            "time-machine",
```

Comment:
This adds `time_machine`, do we need to remove `freezegun` from `install_requires` as well or is it still used somewhere?

Reply:
It's used in core/tests/definition.py so people can freeze time in their SQLMesh tests.

By default, the tests are run locally in DuckDB. 

freezegun seems to be correctly interoperating with DuckDB's default tz handling, so it seems dangerous to switch to something fully tz-aware (which will therefore require explicit tz handling, possibly by the user).

Reply:
So it might make sense to update that too to time-machine but yeah it could be a breaking change and if so makes sense to leave out-of-scope of this PR.